### PR TITLE
Update contributing doc

### PR DESCRIPTION
**Pull request by xiaoyu-work**
## Describe your changes

Update contributing doc

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Format your code by running `pre-commit run --all-files`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: .lintrunner.toml [hunk 1]
```diff
@@ -57,7 +57,7 @@ init_command = [
   'run',
   'pip_init',
   '--dry-run={{DRYRUN}}',
-  '--requirement=requirements-lintrunner.txt'
+  '--requirement=requirements-dev.txt'
```
Comment (guotuofeng) [inline .lintrunner.toml:60]: revert the change

**Comment by guotuofeng** [comment]
The PR LGTM. just not sure why the CI pipeline fails

**Comment by jambayk** [comment]
> The PR LGTM. just not sure why the CI pipeline fails

The bert_cpu_aml has been a bit flaky recently. Looks like we got a bad node in the cluster. Hopefully, it got deallocated during the downtime. 

If the current run fails again due to cached aml jobs, we could try triggerring a fresh one using `azp run`

**Review by guotuofeng — DISMISSED** [review]
could we separate the lintrunner changes in another PR. otherwise, it would have conflicts with https://github.com/microsoft/Olive/pull/659

**Comment by jambayk** [comment]
> could we separate the lintrunner changes in another PR. otherwise, it would have conflicts with #659

@guotuofeng Since Xiaoyu is away, we either need to first merge this PR and update your PR. Or have someone take over this PR to make it compatible with yours.

**Comment by guotuofeng** [comment]
> > could we separate the lintrunner changes in another PR. otherwise, it would have conflicts with #659
> 
> @guotuofeng Since Xiaoyu is away, we either need to first merge this PR and update your PR. Or have someone take over this PR to make it compatible with yours.

Ok, I will merge my PR first then take over this PR

**Review by guotuofeng — APPROVED** [review]

**Review by trajepl — APPROVED** [review]
